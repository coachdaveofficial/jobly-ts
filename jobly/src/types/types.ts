export type Company = {
    handle: string;
    name: string;
    num_employees: number | null;
    description: string;
    logo_url: string | null;
};

export type Job = {
    id: number;
    title: string;
    salary: number | null;
    equity: number | null;
    company_handle: string;
};