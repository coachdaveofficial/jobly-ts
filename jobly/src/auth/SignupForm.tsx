import React, { useState } from 'react'
import './Login.css'
import { SignupFuncProp } from '../types/types'
import { useNavigate } from 'react-router-dom'
import Alert from '../common/Alert'



export default function SignupForm({ signup }: SignupFuncProp) {
    const INITIAL_STATE = {
        username: "",
        password: "",
        firstName: "",
        lastName: "",
        email: ""

    }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const [formErrors, setFormErrors] = useState([]);
    const navigate = useNavigate();

    const handleChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
        const { name, value } = evt.target;
        setFormData((prevData) => ({
            ...prevData,
            [name]: value
        }));
    }

    const handleSubmit = async (evt: React.FormEvent<HTMLFormElement>) => {
        evt.preventDefault();
        let result = await signup(formData);
        if (result.success) {
            navigate('/companies')
        } else {
            setFormErrors(result.error);
        }

    }

    return (
        <>
            <form className='my-2 col-md-8 offset-md-2' onSubmit={handleSubmit}>

                <div className="login-container">
                    <label className='form-label' htmlFor="username"><b>Username</b></label>
                    <input
                        className='form-control'
                        id='username'
                        type="text"
                        placeholder="Enter Username"
                        name="username"
                        value={formData.username}
                        onChange={handleChange}
                        required
                    />

                    <label className='form-label' htmlFor="password"><b>Password</b></label>
                    <input
                        className='form-control'
                        id='password'
                        type="password"
                        placeholder="Enter Password"
                        name="password"
                        value={formData.password}
                        onChange={handleChange}
                        required />

                    <label className='form-label' htmlFor="firstName"><b>First Name</b></label>
                    <input
                        className='form-control'
                        id='firstName'
                        type="text"
                        placeholder="Enter First Name"
                        name="firstName"
                        value={formData.firstName}
                        onChange={handleChange}
                        required />

                    <label className='form-label' htmlFor="lastName"><b>Last Name</b></label>
                    <input
                        className='form-control'
                        id='lastName'
                        type="text"
                        placeholder="Enter Last Name"
                        name="lastName"
                        value={formData.lastName}
                        onChange={handleChange}
                        required />

                    <label className='form-label' htmlFor="email"><b>Email</b></label>
                    <input
                        className='form-control'
                        id='email'
                        type="text"
                        placeholder="Enter Email"
                        name="email"
                        value={formData.email}
                        onChange={handleChange}
                        required />
                        
                    {formErrors.length
                        ? <Alert type="danger" messages={formErrors} />
                        : null}

                    <button className='btn btn-primary' type="submit">Signup</button>

                </div>

                <div className="login-container" style={{ backgroundColor: '#f1f1f1' }}>
                    <button type="button" className="cancelbtn btn btn-danger" onClick={() => navigate('/')}>Cancel</button>
                </div>
            </form>
        </>

    )
}