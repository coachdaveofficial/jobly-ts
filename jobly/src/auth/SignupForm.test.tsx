import React from "react";
import { render } from "@testing-library/react";
import SignupForm from "./SignupForm";
import { MemoryRouter } from "react-router";
import { SignupData } from "../types/types";

const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <SignupForm signup={function (data: SignupData): Promise<{ success: boolean; error?: any; }> {
                throw new Error("Function not implemented.");
            }} />
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it('renders the signup form', () => {

    const { getByLabelText, getByText } = render(<SignupForm signup={function (data: SignupData): Promise<{ success: boolean; error?: any; }> {
        throw new Error("Function not implemented.");
    }} />)
    // Ensure the form elements are rendered
    expect(getByLabelText('Username')).toBeInTheDocument();
    expect(getByLabelText('Password')).toBeInTheDocument();
    expect(getByLabelText('First Name')).toBeInTheDocument();
    expect(getByLabelText('Last Name')).toBeInTheDocument();
    expect(getByLabelText('Email')).toBeInTheDocument();
    expect(getByText('Signup')).toBeInTheDocument();
    expect(getByText('Cancel')).toBeInTheDocument();
});