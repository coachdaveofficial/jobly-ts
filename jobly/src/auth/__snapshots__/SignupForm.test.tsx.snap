// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="my-2 col-md-8 offset-md-2"
  >
    <div
      class="login-container"
    >
      <label
        class="form-label"
        for="username"
      >
        <b>
          Username
        </b>
      </label>
      <input
        class="form-control"
        id="username"
        name="username"
        placeholder="Enter Username"
        required=""
        type="text"
        value="testuser"
      />
      <label
        class="form-label"
        for="password"
      >
        <b>
          Password
        </b>
      </label>
      <input
        class="form-control"
        id="password"
        name="password"
        placeholder="Enter Password"
        required=""
        type="password"
        value="password"
      />
      <label
        class="form-label"
        for="firstName"
      >
        <b>
          First Name
        </b>
      </label>
      <input
        class="form-control"
        id="firstName"
        name="firstName"
        placeholder="Enter First Name"
        required=""
        type="text"
        value=""
      />
      <label
        class="form-label"
        for="lastName"
      >
        <b>
          Last Name
        </b>
      </label>
      <input
        class="form-control"
        id="lastName"
        name="lastName"
        placeholder="Enter Last Name"
        required=""
        type="text"
        value=""
      />
      <label
        class="form-label"
        for="email"
      >
        <b>
          Email
        </b>
      </label>
      <input
        class="form-control"
        id="email"
        name="email"
        placeholder="Enter Email"
        required=""
        type="text"
        value=""
      />
      <button
        class="btn btn-primary"
        type="submit"
      >
        Signup
      </button>
    </div>
    <div
      class="login-container"
      style="background-color: rgb(241, 241, 241);"
    >
      <button
        class="cancelbtn btn btn-danger"
        type="button"
      >
        Cancel
      </button>
    </div>
  </form>
</DocumentFragment>
`;
